modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -607,7 +632,8 @@ protected Assignment getAssignment(User.ID userID, Application.Name applicationN
         Assignment assignment = getAssignment(experimentID, userID, context, userAssignments, bucketList);
         if (assignment == null || assignment.isBucketEmpty()) {
             if (createAssignment) {
-                if (experiment.getState() == Experiment.State.PAUSED) {
+                if (experiment.getState() == Experiment.State.PAUSED ||
+                        !isUserCapForRapidExperimentReached(experiment, rapidExperimentToAssignmentCounts)) {
```

Comment:
@abhishekjain88 As the name of this method changed to isUserCapForRapidExperimentReached(), should it now return TRUE when user cap is reached else FALSE.